lib/data/adapters/activity_isar_adapter.dart

```diff
@@ -0,0 +1,151 @@
+import 'package:injectable/injectable.dart';
+import 'package:movna/data/adapters/base_adapter.dart';
+import 'package:movna/data/models/activity_model.dart';
+import 'package:movna/domain/entities/activity.dart';
+import 'package:movna/domain/entities/gps_coordinates.dart';
+import 'package:movna/domain/entities/location.dart';
+import 'package:movna/domain/entities/track_point.dart';
+import 'package:movna/domain/entities/track_segment.dart';
+
+@injectable
+class ActivityIsarAdapter extends BaseAdapter<Activity, ActivityModel> {
+  ActivityIsarAdapter(this.trackSegmentIsarAdapter);
+
+  TrackSegmentIsarAdapter trackSegmentIsarAdapter;
+
+  @override
+  Activity modelToEntity(ActivityModel m) {
+    return Activity(
+      name: m.name,
+      sport: m.sport,
+      // TODO : convert startTime and stopTime back to their original timezone
```

Comment:
do not use timezones, always use UTC and then the presentation will account for timezones when displaying information to the user, all dates operations should be made using UTC dates

Reply:
Yes, UTC should be used to store the dates in the database (even more since I use the startTime as unique ID), but the Datetimes get stored as UTC, and are loaded in localtime by Isar automatically. The problem with that is that if one does an activity at 12h in april, it will be marked as done at 11h (or 13h I don't remember) after the DST change in may or june. The phenomenon is even worse if the user wants to see its activities when they're in another country, it will be marked as done at whatever time it was in their local time when they did the activity in their country of origin.

I think this behavior is ok for now, but it is definitely an issue to address in the future.

Reply:
ok I get what you mean, in this particular case we don't want to display the time using current timezone but actually using the activity's timezone so that the user knows at what time of day they actually exercised.

What we could do is simply store the activity's timezone in isar so that the presentation can make the correct adjustments (we could imagine a setting to either display at current timezone or activity timezone)

Reply:
also, if you create a TODO, create an issue and link it in the TODO so that we can keep track of that

Reply:
There are also timezone-aware Datetime classes, but I think the best option in this case is to create a custom `Datetime` class, that could either store :
- `Date`, `TimeOfDay`, `localTimeOffset`
- `DateTime`, `localTimeOffset`
I don't really have an answer there

Reply:
the second solution seems fine to me honestly, `localTimeOffset` should be a double as there are 45" and 30" offsets but apart from that I'm good with this soltion

Reply:
Or we could simply decide to let things as-are, and in the HMI display the local timezone. To keep going with my example, I would see "Activity done at 5am Eastern Standard Time".

I have dealt with timezones before and it is a complete nightmare, the less we do things about it, the better

Reply:
I say let's just store the `localTimeOffset` and if it happens to be too much of a trouble we'll just ignore it / delete it but I'd rather try to do this best we can